### PR TITLE
fix: table save row button onclick function

### LINKED ISSUE #35146: [Task]: Fix flakiness of cypress/e2e/Regression/ClientSide/OneClickBinding/TableWidget/mysql_spec.ts
### SubTasks

Entire spec is skipped from July 6, 2023 Ref: https://github.com/appsmithorg/appsmith/commit/f24ecc2473b86b8f87f1e54e6d1b8a87383a1137

### PR DESCRIPTION
**Pull request by AmanAgarwal041**
## Description
The `onClick` method was attached to both button and parent child : https://github.com/appsmithorg/appsmith/blob/release/app/client/src/widgets/ButtonWidget/component/index.tsx#L204-L232
Hence, whenever we clicked on the button using the UI, since there was a check of loading state inside the onclick handler it was not triggering twice as before running the onclick of parent it was updating the reference value of loading inside it.
But if we trigger the button click using cypress or the `document.querySelector` from console, it used to execute the addnewrow action twice, thereby creating two rows. Hence added a stopPropagation function in the onclick handler, which restricts the onclick to propagate to the parent.

Fixes #35146 

EE runs - https://github.com/appsmithorg/appsmith-ee/pull/4818
_or_  
Fixes `Issue URL`
> [!WARNING]  
> _If no issue exists, please create an issue first, and check with the maintainers if the issue is valid._

## Automation

/ok-to-test tags="@tag.Widget, @tag.Datasource, @tag.Binding"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/10301142371>
> Commit: 40a81cb92810f1ceca334429d135b8832aef1abf
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=10301142371&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Widget, @tag.Datasource, @tag.Binding`
> Spec:
> <hr>Thu, 08 Aug 2024 12:44:05 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
	- Improved user experience by refining the click event handling in the Add New Row banner, preventing unintended interactions.

- **Tests**
	- Enhanced the Cypress test suite for the Table Widget, improving organization and readability while ensuring accurate verification of the one-click binding feature.
	- Updated limited tests to include multiple components related to ClientSide functionality, ensuring more comprehensive testing coverage.
	- Added a check for the filterable property of the table widget in Cypress tests, enhancing test robustness and accuracy. 
	- Improved the locator method for the "filterable" input in tests to enhance maintainability and resilience to UI changes.

### CODE REVIEW

File: app/client/cypress/e2e/Regression/ClientSide/OneClickBinding/TableWidget/mysql_spec.ts [hunk 1]
```diff
@@ -14,88 +14,91 @@ import EditorNavigation, {
...
-    });
 
-    agHelper.GetNClick(oneClickBindingLocator.connectData);
+      agHelper.GetNClick(oneClickBindingLocator.connectData);
 
-    assertHelper.AssertNetworkStatus("@postExecute");
+      assertHelper.AssertNetworkStatus("@postExecute");
 
-    agHelper.Sleep(2000);
+      agHelper.Sleep(2000);
```
Comment (ApekshaBhosale) [inline app/client/cypress/e2e/Regression/ClientSide/OneClickBinding/TableWidget/mysql_spec.ts:38]: remove sleep please

File: app/client/cypress/e2e/Regression/ClientSide/OneClickBinding/TableWidget/mysql_spec.ts [hunk 2]
```diff
@@ -14,88 +14,91 @@ import EditorNavigation, {
...
 
-    table.UpdateTableCell(0, 2, `{{}"key":"oneClick"}`);
-    table.UpdateTableCell(0, 3, 36);
-    table.UpdateTableCell(0, 4, "2023-07-03 15:30:00", false, true);
+      table.UpdateTableCell(0, 2, `{{}"key":"oneClick"}`);
+      table.UpdateTableCell(0, 3, 36);
+      table.UpdateTableCell(0, 4, "2023-07-03 15:30:00", false, true);
 
-    agHelper.Sleep(2000);
+      agHelper.Sleep(2000);
```
Comment (ApekshaBhosale) [inline app/client/cypress/e2e/Regression/ClientSide/OneClickBinding/TableWidget/mysql_spec.ts:59]: remove sleep

File: app/client/cypress/e2e/Regression/ClientSide/OneClickBinding/TableWidget/mysql_spec.ts [hunk 3]
```diff
@@ -14,88 +14,91 @@ import EditorNavigation, {
...
+      agHelper.TypeText(table._searchInput, "One Click Config");
 
-    assertHelper.AssertNetworkStatus("@postExecute");
+      assertHelper.AssertNetworkStatus("@postExecute");
 
-    agHelper.AssertElementExist(table._bodyCell("One Click Config"));
+      agHelper.AssertElementExist(table._bodyCell("One Click Config"));
 
-    agHelper.Sleep(1000);
+      agHelper.Sleep(1000);
```
Comment (ApekshaBhosale) [inline app/client/cypress/e2e/Regression/ClientSide/OneClickBinding/TableWidget/mysql_spec.ts:71]: remove sleep

File: app/client/cypress/e2e/Regression/ClientSide/OneClickBinding/TableWidget/mysql_spec.ts [hunk 4]
```diff
@@ -14,88 +14,91 @@ import EditorNavigation, {
...
 
-    table.EditTableCell(0, 1, "Bindings", false);
-    table.EditTableCell(0, 4, "2023-07-03 15:30:00", false);
+      table.EditTableCell(0, 1, "Bindings", false);
+      table.EditTableCell(0, 4, "2023-07-03 15:30:00", false);
 
-    agHelper.Sleep(1000);
+      agHelper.Sleep(1000);
```
Comment (ApekshaBhosale) [inline app/client/cypress/e2e/Regression/ClientSide/OneClickBinding/TableWidget/mysql_spec.ts:76]: remove sleep

File: app/client/cypress/e2e/Regression/ClientSide/OneClickBinding/TableWidget/mysql_spec.ts [hunk 5]
```diff
@@ -14,88 +14,91 @@ import EditorNavigation, {
...
 
-    assertHelper.AssertNetworkStatus("@postExecute");
+      assertHelper.AssertNetworkStatus("@postExecute");
 
-    assertHelper.AssertNetworkStatus("@postExecute");
+      assertHelper.AssertNetworkStatus("@postExecute");
 
-    agHelper.Sleep(500);
-    agHelper.ClearNType(table._searchInput, "Bindings");
+      agHelper.Sleep(500);
```
Comment (ApekshaBhosale) [inline app/client/cypress/e2e/Regression/ClientSide/OneClickBinding/TableWidget/mysql_spec.ts:86]: remove sleep

File: app/client/cypress/e2e/Regression/ClientSide/OneClickBinding/TableWidget/mysql_spec.ts [hunk 6]
```diff
@@ -14,88 +14,91 @@ import EditorNavigation, {
...
+      agHelper.ClearNType(table._searchInput, "Bindings");
 
-    assertHelper.AssertNetworkStatus("@postExecute");
+      assertHelper.AssertNetworkStatus("@postExecute");
 
-    agHelper.Sleep(2000);
+      agHelper.Sleep(2000);
```
Comment (ApekshaBhosale) [inline app/client/cypress/e2e/Regression/ClientSide/OneClickBinding/TableWidget/mysql_spec.ts:91]: remove sleep

File: app/client/cypress/e2e/Regression/ClientSide/OneClickBinding/TableWidget/mysql_spec.ts [hunk 7]
```diff
@@ -14,88 +14,91 @@ import EditorNavigation, {
...
+      agHelper.AssertElementExist(table._bodyCell("Bindings"));
 
-    agHelper.ClearNType(table._searchInput, "One Click Config");
+      agHelper.ClearNType(table._searchInput, "One Click Config");
 
-    assertHelper.AssertNetworkStatus("@postExecute");
+      assertHelper.AssertNetworkStatus("@postExecute");
 
-    agHelper.Sleep(2000);
+      agHelper.Sleep(2000);
```
Comment (ApekshaBhosale) [inline app/client/cypress/e2e/Regression/ClientSide/OneClickBinding/TableWidget/mysql_spec.ts:99]: remove sleep

**Review by ApekshaBhosale — CHANGES_REQUESTED** [review]
remove all static wait ie agHelper.Sleep and the execute this spec multiple times on EE and CE that will give you clear idea if removing static wait has removed all flakiness. @AmanAgarwal041

File: app/client/cypress/e2e/Regression/ClientSide/OneClickBinding/TableWidget/mysql_spec.ts (hunk 7 above)
Comment (ApekshaBhosale) [reply app/client/cypress/e2e/Regression/ClientSide/OneClickBinding/TableWidget/mysql_spec.ts:99]: @coderabbitai please suggest not to use this agHelper.Sleep on every PR

Comment (sagar-qa007) [reply app/client/cypress/e2e/Regression/ClientSide/OneClickBinding/TableWidget/mysql_spec.ts:99]: @coderabbitai Please put comments on PR. Add learning as well for future use cases.

Broken rule: "- Avoid using agHelper.Sleep, this.Sleep in any file in code."

Comment (sagar-qa007) [reply app/client/cypress/e2e/Regression/ClientSide/OneClickBinding/TableWidget/mysql_spec.ts:99]: @codetabbitai can you add comments now in this PR?

File: app/client/cypress/e2e/Regression/ClientSide/OneClickBinding/TableWidget/mysql_spec.ts (hunk 1 above)
Comment (AmanAgarwal041) [reply app/client/cypress/e2e/Regression/ClientSide/OneClickBinding/TableWidget/mysql_spec.ts]: Done

File: app/client/cypress/e2e/Regression/ClientSide/OneClickBinding/TableWidget/mysql_spec.ts (hunk 7 above)
Comment (AmanAgarwal041) [reply app/client/cypress/e2e/Regression/ClientSide/OneClickBinding/TableWidget/mysql_spec.ts]: Removed sleep!

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/columnTypes/Select1_spec.ts [hunk 8]
```diff
@@ -363,7 +363,12 @@ describe(
       PageLeftPane.switchSegment(PagePaneSegment.UI);
       cy.openPropertyPane("tablewidgetv2");
       cy.editColumn("step");
-      cy.get(".t--property-control-filterable input").click({ force: true });
+      cy.get(".t--property-control-filterable input").then(($checkbox) => {
+        // Check if the checkbox is checked
```
Comment (ApekshaBhosale) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/columnTypes/Select1_spec.ts:367]: @AmanAgarwal041 we have helper for this. @NandanAnantharamu can you please suggest the helper please?

**Review by ApekshaBhosale — CHANGES_REQUESTED** [review]

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/columnTypes/Select1_spec.ts (hunk 8 above)
Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/columnTypes/Select1_spec.ts:367]: @AmanAgarwal041 please use CheckUncheck(srelector,flag) its defined in agHelper

Comment (AmanAgarwal041) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/columnTypes/Select1_spec.ts]: Done

**Comment by AmanAgarwal041** [comment]
EE PR : https://github.com/appsmithorg/appsmith-ee/pull/4818

File: app/client/cypress/limited-tests.txt [hunk 9]
```diff
@@ -1,5 +1,6 @@
 # To run only limited tests - give the spec names in below format:
-cypress/e2e/Regression/ClientSide/Templates/Fork_Template_spec.js
+cypress/e2e/Regression/ClientSide/OneClickBinding/TableWidget/mysql_spec.ts
```
Comment (ApekshaBhosale) [inline app/client/cypress/limited-tests.txt:2]: revert this file please

**Review by ApekshaBhosale — CHANGES_REQUESTED** [review]

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/columnTypes/Select1_spec.ts [hunk 10]
```diff
@@ -363,6 +363,7 @@ describe(
       PageLeftPane.switchSegment(PagePaneSegment.UI);
       cy.openPropertyPane("tablewidgetv2");
       cy.editColumn("step");
+      _.agHelper.CheckUncheck(".t--property-control-filterable input", true);
```
Comment (ApekshaBhosale) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/columnTypes/Select1_spec.ts:366]: add this to locator file if this doesn't exists already and use .t--property-control-filterable input

**Review by ApekshaBhosale — CHANGES_REQUESTED** [review]

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/columnTypes/Select1_spec.ts (hunk 10 above)
Comment (AmanAgarwal041) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/columnTypes/Select1_spec.ts]: Updated the code.

File: app/client/cypress/limited-tests.txt (hunk 9 above)
Comment (AmanAgarwal041) [reply app/client/cypress/limited-tests.txt]: Done

**Review by ApekshaBhosale — APPROVED** [review]

**Pull request merged**